tests/components/calendar/test_init.py

```diff
@@ -553,3 +557,43 @@ async def test_list_events_missing_fields(hass: HomeAssistant) -> None:
             blocking=True,
             return_response=True,
         )
+
+
+@pytest.mark.parametrize(
+    "frozen_time", ["2023-06-22 10:30:00+00:00"], ids=["frozen_time"]
+)
+@pytest.mark.parametrize(
+    ("service_data"),
+    [
+        {
+            "start_date_time": "2023-06-22T04:30:00-06:00",
+            "end_date_time": "2023-06-22T04:30:00-06:00",
+        },
+        {
+            "start_date_time": "2023-06-22T04:30:00",
+            "end_date_time": "2023-06-22T04:30:00",
+        },
+        {"start_date_time": "2023-06-22", "end_date_time": "2023-06-22"},
+        {"start_date_time": "2023-06-22 10:00:00", "duration": "0"},
+    ],
+)
+async def test_list_events_service_same_dates(
+    hass: HomeAssistant,
+    service_data: dict[str, str],
+) -> None:
+    """Test listing events from the service call using the same start and end time."""
+
+    with pytest.raises(
+        ServiceValidationError, match="End date must be after start date"
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_GET_EVENTS,
+            target={"entity_id": ["calendar.calendar_1"]},
```

Comment:
Target isn't needed if `entity_id` is included in the call data parameter and vice versa.

Reply:
Oops!